io/io/inc/ROOT/RIoUring.hxx

```diff
@@ -56,6 +61,91 @@ public:
       static const bool available = RIoUring::CheckIsAvailable();
       return available;
    }
+
+   /// Access the raw io_uring instance.
+   struct io_uring *GetRawRing() {
+      return &fRing;
+   }
+
+   /// Basic read event composed of RIOVec IO data and a target file descriptor.
+   struct RReadEvent {
+      RRawFile::RIOVec* fIoVec = nullptr;
+      int fFileDes = -1;
+   };
+
+   /// Submit a number of read events and wait for completion.
+   void SubmitReadsAndWait(std::vector<RReadEvent>& readEvents) {
```

Comment:
Perhaps better to use an `std::span<ReadEvent>` (include `ROOT/RSpan.hxx`)

Reply:
I changed this to a pointer and length in 592ef14 following our discussion